src/DotLiquid/Util/ObjectExtensionMethods.cs

```diff
@@ -52,8 +54,10 @@ public static bool BackCompatSafeTypeInsensitiveEqual(this object value, object
         {
             // NOTE(daviburg): Historically testing for equality cross integer and string boundaries resulted in not equal.
             // This ensures we preserve the behavior.
-            if (value is string && !(otherValue is char) || otherValue is string && !(value is char))

```

Comment:
Could it be this condition was simply missing the condition where both types are equal, strings in this case?

Reply:
No. My test passes without type equality as well, just shorter to test that first. 
```
                if ((_numericTypes.Contains(aType) ^ _numericTypes.Contains(bType)))
                {
                    return false;
                }
```
All conditional tests pass.